internal/db/schema/migrations/oss/postgres/70/02_sessions.up.sql

```diff
@@ -0,0 +1,189 @@
+    for each row execute procedure cancel_session_with_null_target_address_fk();
+
+  create table session_host_set_host (
+    session_id wt_public_id primary key,
+    host_set_id wt_public_id,
+    host_id wt_public_id,
+    -- the following foreign key constraints for host_set_id & host_id is set to null on a delete
+    -- because we want to be able to invoke a trigger to call cancel_session
+    -- when the references column is deleted
+    -- rather than canceling a session for any reason that may cause a row to be deleted.
+    constraint host_set_fkey foreign key (host_set_id)
+        references host_set (public_id)
+        on delete set null
+        on update cascade,
+    constraint host_fkey foreign key (host_id)
+        references host (public_id)
+        on delete set null
+        on update cascade,
+    constraint session_fkey foreign key (session_id)
+        references session (public_id)
+        on delete cascade
+        on update cascade,
+    constraint session_id_host_set_id_host_id_uq
+        unique(session_id, host_set_id, host_id)
+  );
+  comment on table session_host_set_host is
+    'session_host_set entries represent a session that is using a Host Set.';
+
+  create trigger immutable_columns before update on session_host_set_host
+    for each row execute procedure immutable_columns('session_id');
+
+  create function cancel_session_with_null_host_source_fk() returns trigger
+  as $$
+  begin
+    if new.host_set_id is null or new.host_id is null then
+      perform cancel_session(new.session_id);
+      delete from session_host_set_host where session_id = new.session_id;
+    end if;
+    return new;
+  end;
+  $$ language plpgsql;
+  
+  create trigger cancel_session_with_null_host_source_fk after update of host_set_id, host_id on session_host_set_host
+    for each row execute procedure cancel_session_with_null_host_source_fk();
+
+  drop view session_list;
+  alter table session
+    drop constraint session_host_id_fkey,
+    drop constraint session_host_set_id_fkey,
+    drop column host_id,
+    drop column host_set_id
+  ;
+
+  -- Replaces trigger from 44/04_sessions.up.sql
+  drop trigger insert_session on session;
+  create or replace function insert_session() returns trigger
+  as $$
+  begin
+    case
+      when new.user_id is null then
+        raise exception 'user_id is null';
+      when new.target_id is null then
+        raise exception 'target_id is null';
+      when new.auth_token_id is null then
+        raise exception 'auth_token_id is null';
+      when new.project_id is null then
+        raise exception 'project_id is null';
+      when new.endpoint is null then
+        raise exception 'endpoint is null';
+    else
+    end case;
+    return new;
+  end;
+  $$ language plpgsql;
+
+  create trigger insert_session before insert on session
+    for each row execute procedure insert_session();
+
+  -- Replaces view from 59/01_target_ingress_egress_worker_filters.up.sql
```

Comment:
Please add a comment in `59/01_target_ingress_egress_worker_filters.up.sql` about the view being updated